### PR TITLE
Timer

### PR DESCRIPTION
**Pull request by Sarahtonein**
Removed vote_timer function.

Added Discord task to routinely check for concluded proposals.

Added corresponding function in gov/proposals.py which handles the logic for the Discord task.

Added JSON file to track ongoing votes. These are purged when a vote has concluded and the result has been posted.

Removed leading spaces for the proposal information posted to the forum channel, the formatting of this message could be thrown off previously.

### CODE REVIEW

**Comment by Sarahtonein** [comment]
Closing this for now.

**Comment by Balugrizzly** [comment]
> Closing this for now.

why?

**Comment by Sarahtonein** [comment]
> > Closing this for now.
> 
> why?

Working on merge issues and didn't want to leave it hanging

**Comment by Balugrizzly** [comment]
What do you mean by hanging?

**Comment by Sarahtonein** [comment]
> What do you mean by hanging?

Sorry to be clearer I meant I didn't want to leave the PR open when I was going to make other changes first. Am going to try have this one ready today anyway now :)

**Comment by Sarahtonein** [comment]
> > What do you mean by hanging?
> 
> Sorry to be clearer I meant I didn't want to leave the PR open when I was going to make other changes first. Am going to try have this one ready today anyway now :)

Ready for this one to be reviewed and merged when ready :)

File: events/events.py
Comment (Balugrizzly) [inline events/events.py:1]: Starting backgrounds tasks in the "on_ready" function can run into race conditions.
Why have you not used the pre existing structure that i have set up for background tasks?

Comment (Balugrizzly) [inline events/events.py:1]: I also see you too use backgrounds tasks so this is even more confusing now

File: tasks/tasks.py [hunk 1]
```diff
@@ -79,4 +83,113 @@ async def check_events(bot: commands.Bot) -> None:
...
+                    f"Unable to find the thread with id: {proposal_id} in the channel: {channel.name}"
+                )
+                continue
+
+            message = await thread.fetch_message(int(proposal_id))
+            if not message:
+                logger.error(
+                    f"Unable to find the message with id: {proposal_id} in the thread: {thread.name}"
+                )
+                continue
```
Comment (Balugrizzly) [inline tasks/tasks.py:127]: :) perhaps consider a loop or method chaining here

File: events/events.py
Comment (Sarahtonein) [reply events/events.py:1]: > Starting backgrounds tasks in the "on_ready" function can run into race conditions. Why have you not used the pre existing structure that i have set up for background tasks?

Can you elaborate on the pre existing structure you are referring to? 

Can jump on VC to discuss if easier.

File: tasks/tasks.py (hunk 1 above)
Comment (Sarahtonein) [reply tasks/tasks.py:127]: > :) perhaps consider a loop or method chaining here

Will look at this today 👍🏼

Comment (Sarahtonein) [reply tasks/tasks.py:127]: > :) perhaps consider a loop or method chaining here

Cannot do method chaining as I am using fetch_message which is async

Comment (Sarahtonein) [reply tasks/tasks.py]: Have reduced the if / elif statements by using a dict and a for loop. 

I am not so sure I have addressed your other point of using a loop, there is already a for loop in use.

Comment (Balugrizzly) [reply tasks/tasks.py]: i meant for the whole if else blocks but all good

File: events/events.py
Comment (Balugrizzly) [reply events/events.py:1]: vc sounds good to me usually time 222222 UTCCCC :)

**Review by Balugrizzly — APPROVED** [review]
Looks good so far assuming we do VC

**Review by Balugrizzly — APPROVED** [review]
Looks good

**Pull request merged**